src/types.js

```diff
@@ -208,6 +194,51 @@ export type Outbox = $ReadOnly<{|
   >,
 |}>;
 
+export type PmOutbox = {|
+  ...OutboxBase,
+
+  ...SubsetProperties<PmMessage, {||}>,
+|};
+
+export type StreamOutbox = {|
+  ...OutboxBase,
+
+  // TODO(#3764): Make stream_id required.  Needs a migration to drop
+  //   StreamOutbox values that lack it; that'll be fine once the
+  //   release that adds it has been out for a few weeks. (Also drop
```

Comment:
There's another key step missing before we can do this: we need to start actually populating this property in the values we create for `state.outbox` :wink: See a1fad7ca9 for comparison.